File: src/main/java/com/thss/chat/domain/security/model/vo/JwtToken.java
package com.thss.chat.domain.security.model.vo;

import lombok.AllArgsConstructor;
import org.apache.shiro.authc.AuthenticationToken;

/**
 * @author yjy
 * @emial yyyejunyu@gmail.com
 * @date 2023/12/15
 * @description
 **/
@AllArgsConstructor
public class JwtToken implements AuthenticationToken {
    private String jwt;

    @Override
    public Object getPrincipal() {
        return jwt;
    }

    @Override
    public Object getCredentials() {
        return jwt;
    }
}

File: src/main/java/com/thss/chat/domain/security/model/vo/JwtToken.kt
package com.thss.chat.domain.security.model.vo

import org.apache.shiro.authc.AuthenticationToken

/**
 * @author yjy
 * @emial yyyejunyu@gmail.com
 * @date 2023/12/15
 * @description Token对象存用户id
 **/
data class JwtToken(private val jwt: String) : AuthenticationToken {
    override fun getPrincipal(): Any {
        return jwt
    }

    override fun getCredentials(): Any {
        return jwt
    }

}
